### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by mikee47**
General review and update to documentation

**Smart quotes**

These are the 'pretty' versions of regular quotes which can infect code and documentation due to a 'smart quote' virus embedded into many modern editors.
Their purpose is to make documentation (e.g. web pages) look pretty. That's it.

Their presence in C/C++ code is a syntax errror. In documentation, not so much.
However, if they appear in quoted code snippets then these won't work when pasted.

I think it best therefore to avoid them completely, so have all been converted to standard 'dumb' quotes.

Note that Sphinx will automatically convert regular quotes into their curly equivalents so there is no need whatsoever to have these in the source.

TODO (please comment if you have any thoughts on these!):

* [x] Is espconn still unsupported by lwip2? Don't know: just add note.
* [x] Windows .cmd install scripts are quite useable, just needs an `export.cmd` and documenting. We then don't need the custom sming choco scripts: we should remove those from documentation. Bit more than that: separate PR.
* [x] Is there a way to check/fix use of smart quotes automatically? Nope. I'll write one.
* [x] Add notes on debugging in vscode - already there, just not as involved as Eclipse
* [x] Is eclipse still a good development environment? I always found it slow, complicated and clunky, but maybe that's changed?  Is the documentation still correct for current versions?
* [x] Lots of duplication in eclipse debugging pages, merge this into single document

### CODE REVIEW

**Comment by mikee47** [comment]
@slaff Maybe leave this one open for general documentation fixes until we're ready for release?

**Comment by mikee47** [comment]
Test build for review https://smingdev.readthedocs.io/en/fix-documentation-fixes/

**Comment by slaff** [comment]
@mikee47 I am planning to create the new 5.2.0 release at the end of next week. Do you need more time to make all changes that you have planned or this works for you?

**Comment by mikee47** [comment]
@slaff That'll be fine. I'm aiming to finish this PR today sometime. Also got some rework for CI build caching due shortly.

**Comment by mikee47** [comment]
@slaff OK so documentation is *almost* there but still keep finding stuff. If I find anything which needs a bit of research I'll add it as a TODO but otherwise it's fixing errors, inconsistencies and duplication.

**Comment by slaff** [comment]
@mikee47 thank you for your hard work. This release is going to have the best, for now, documentation and code improvements. Mainly thanks to you :)

**Comment by mikee47** [comment]
It surprises me how much I (didn't) know: clang, llvm, lldb, macos, timezones, ccache, CI caching/artifacts,.... I've learned a load in the past while!

**Comment by slaff** [comment]
> We then don't need the custom sming choco scripts: we should remove those from documentation.

Lets keep it for this release, and announce that choco is deprecated. We will remove it for release v.6.0

> Is eclipse still a good development environment? I always found it slow, complicated and clunky, but maybe that's changed? Is the documentation still correct for current versions?

Old guys like me are still using it :) I will look at the documentation but the last time it was still ok and Eclipse hasn't changed much in the last years.

**Comment by mikee47** [comment]
> We then don't need the custom sming choco scripts: we should remove those from documentation.
>> Lets keep it for this release, and announce that choco is deprecated. We will remove it for release v.6.0

Spoke too soon... Forgotten what a PITA Windows was. Path lengths, etc, etc, etc. Also having attempted a scripted winget installation the experience is much better with choco, so thankfully we can stick with that.

To recap, at present choco.sming does this, in order:

- Install git
- Install python (version 3.9.1 or 3.10)
- Install cmake
- Unpack mingw from SmingTools repo. and update system path
- Install esp8266 toolchain ***
- Clone sming (master branch)

The item *** is out of date now so I'm thinking we remove that and update python to include 3.12.
The user then runs `sming\Tools\install.cmd` to install architecture-specific toolchains,
none of which requires updating the system path so is much easier to handle.

This also puts the system in a similar state to how it is during a CI run.

File: Sming/Core/SmingLocale.h [hunk 1]
```diff
@@ -36,13 +36,13 @@
...
-	"janvier\0f�vrier\0mars\0avril\0mai\0juin\0juillet\0ao�t\0septembre\0octobre\0novembre\0d�cembre"
+	"janvier\0février\0mars\0avril\0mai\0juin\0juillet\0août\0septembre\0octobre\0novembre\0décembre"
 #define LOCALE_DAY_NAMES "dimanche\0lundi\0mardi\0mercredi\0jeudi\0vendredi\0samedi"
 
 #elif LOCALE == LOCALE_DE_DE
 
 #define LOCALE_MONTH_NAMES                                                                                             \
-	"Januar\0Februar\0M�rz\0April\0Mai\0Juni\0Juli\0August\0September\0Oktober\0November\0Dezember"
+	"Januar\0Februar\0März\0April\0Mai\0Juni\0Juli\0August\0September\0Oktober\0November\0Dezember"
 #define LOCALE_DAY_NAMES "Sonntag\0Montag\0Dienstag\0Mittwoch\0Donnerstag\0Freitag\0Samstag"
```
[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Core/SmingLocale.h:46]: I see what happened:

#1549 Initial commit as ascii e9
#1715 replaced with UTF-8 'replacement character' ef bf bd
Correct UTF-8 code is ce a9

[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Core/SmingLocale.h:46]: I'm thinking this should *probably* be the original e9, so not UTF-8 encoded. But that leaves it vulnerable to similar conversion problems in the future unless there's some way to mark the file or represent these characters code immutably.

[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Core/SmingLocale.h]: I've removed this to a separate PR as it's a code change/bugfix.

**Review by slaff — COMMENTED** [review]
Looks great! Noticed only one small typo.

File: docs/source/about.rst [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
-The project was started in 2015 and is actively developed.
 
 Arduino
 -------
+
 Arduino is an open-source electronics platform based on easy-to-use hardware and software.
-Sming is compatible with (most) standard `Arduino libraries <https://sming.readthedocs.io/en/latest/libraries.html>`_,
-which means that you can use any popular hardware in few lines of code.
+Sming is provided with a curated set of :doc:`/libraries`, including many standard Arduino-style ones.
+Lost of popular hardware can be used in a few lines of code.
```
[the remark this answers is not in the document]
Comment (mikee47) [reply docs/source/about.rst]: Eeek. Can't believe I wrote that! `Many popular...` :-)

**Pull request merged**